setup.py

```diff
@@ -76,8 +93,32 @@
         include_package_data=True,
         extras_require={
             'alldeps': (
-                'numpy >= {}'.format(NUMPY_MIN_VERSION),
-                'scipy >= {}'.format(SCIPY_MIN_VERSION),
+                f'numpy>={NUMPY_MIN_VERSION}',
+                f'scipy>={SCIPY_MIN_VERSION}',
+                f'joblib>={JOBLIB_MIN_VERSION}',
+                f'threadpoolctl>={THREADPOOLCTL_MIN_VERSION}'
+            ),
+            'examples': (
+                f'matplotlib>={MATPLOTLIB_MIN_VERSION}',
+                f'scikit-image>={SCIKIT_IMAGE_MIN_VERSION}',
+                f'pandas>={PANDAS_MIN_VERSION}',
+                f'seaborn>={SEABORN_MIN_VERSION}',
+                f'Pillow>={PILLOW_MIN_VERSION}',
+                f'memory_profiler>={MEMORY_PROFILER_MIN_VERSION}'
+            ),
+            'tests': (
+                f'pytest>={PYTEST_MIN_VERSION}',
+                f'pytest-cov>={PYTEST_COV_MIN_VERSION}',
+                f'Cython>={CYTHON_MIN_VERSION}',
```

Comment:
I like the way this is organised. My only doubt is about `Cython`. Basically we need it for building scikit-learn rather than `tests` but I am not really sure how to handle that.

I would prefer a second pair of eye on this @jeremiedbb @thomasjpfan @jnothman @NicolasHug

Reply:
ping @rth

Reply:
Yes, Cython is a build dependency that's already handled by `pyproject.toml`, so it doesn't need to appear there.

Also can we just put the version as string directly instead of using variables with templating?
```
            'tests': (
                f'pytest>=3.3.0',
                f'pytest-cov>=xxx',
```
the reason we did that for scipy and numpy was to allow dynamically change it based on the Python implementation (and because it's used in other parts of setup.py) that doesn't need we need to do it for all other packages.